rir/src/interpreter/interp.c

```diff
@@ -1216,6 +1238,17 @@ INSTRUCTION(stvar_) {
         CLEAR_FRAME_CHANGED(env);
 }
 
+INSTRUCTION(stvar2_) {
+    SEXP sym = readConst(ctx, pc);
+    int wasChanged = FRAME_CHANGED(env);
+    SLOWASSERT(TYPEOF(sym) == SYMSXP);
+    SEXP val = ostack_pop(ctx);
+    INCREMENT_NAMED(val);
+    setVar(sym, val, ENCLOS(env));
+    if (!wasChanged)
+        CLEAR_FRAME_CHANGED(env);
```

Comment:
the FRAME_CHANGED keeps track of whether some non-local change to the env happened. Here this is the case (super assign is non local). Also the change happens on ENCLOS(env). So you can just delete lines 124[3,8,9]